[javentankangle] iP

javentankangle opened the pull request:
# Javen - Task Manager  

Javen is a simple task management application that allows you to add, delete, and list tasks easily. It supports various types of tasks too!

## Features  
- 📝 **Add tasks to your list**  
- ❌ **Delete tasks from the list**  
- 📜 **View all tasks** 
- 💾 **Save tasks automatically** 
- 🕒 **Manage deadlines and events** 

## Tasks Checklist ✅  
- [x] Implement task storage  
- [ ] Add priority levels  
- [ ] Improve UI  


## Installation  
1. Download the latest release: [Javen.jar](https://github.com/javentankangle/ip/releases/download/v0.1/javen.jar)  
2. Double click it and let it work for you :)



**This is a sneak peek of Javen!**
```Ruby
public static void main(String[] args) {
   System.out.println("HiI'm Javen\What do you need help with?"); 
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

manlymonk committed:
rename main class

manlymonk committed:
add greetings

manlymonk committed:
adjust greetings

manlymonk committed:
add echo function

manlymonk committed:
add list item function

manlymonk committed:
add new Task Class

manlymonk committed:
add mark unmark functions in Task class

manlymonk committed:
add mark and unmark function

manlymonk committed:
created inherited class ToDo,Deadline, Event

manlymonk committed:
updated toString function of Task Class

manlymonk committed:
updated echo functions and include Inherited Classes

manlymonk committed:
updated automated testing files:expected.txt, input.txt, runtest.bat

manlymonk committed:
added error checking and updated input.txt and expected.txt

manlymonk committed:
fix ui and updated input/expected.txt

manlymonk committed:
add error handling for mark function

manlymonk committed:
updated input.txt and expected.txt

manlymonk committed:
add delete function

manlymonk committed:
updated expected.txt and input.txt for error checking of delete function

manlymonk committed:
Add Java coding standard

Java coding standard are not applied

Following java coding standard helps code to be more readable

Let's apply java coding standard on all the classes

manlymonk committed:
Add save feature to project Javen
Javen.java: added saveTask and loadTask method
Task.java: allow task to be serializable object

Task list are not saved locally but only per run

Let's allow the task list to be saved locally on hard drive so that task list will not be lost after each run

Task are made serializable so that it can be easily loaded after each save to hard drive

manlymonk committed:
Merge branch 'branch-Level-7'

manlymonk committed:
Add java header comments to all methods in Javen.java and Task.java

Java header comments are not included

Java header comments are included as it helps reader understand methods better

Let's add java header comments to all methods in Javen.java and Task.java

manlymonk committed:
Follow java coding standard

Java coding standard are not used

Java coding standard should be used as it helps readability

Java coding standard is implented in Javen.java

manlymonk committed:
Add searchTask feature to allow users to search for task using keywords, updated error testing

User is unable to serach for tasks using keywords

Searching tasks for keywords is a useful feature for users

Let's include a searchTask feature for users

manlymonk committed:
Merge branch 'A-JavaDoc'

manlymonk committed:
Merge branch 'A-CodingStandard'

manlymonk committed:
Merge tag 'Level-9'

manlymonk committed:
Javen.java: Add parseDateTime function
Event.java: Change constructor to take in LocalDateTime instead of String
Deadline.java: Change constructor to take in LocalDateTime instead of String

User input dates are read as string and not fixed

User input dates should be read as date time and should be easily read and understandable

Let's parse the dates and make it into LocalDateTime and format the dates nicely for users to read

manlymonk committed:
Add new error testing

Error testing files does not include parseDateTime function

Testing for parseDateTime function is necessary to ensure no breakdown

Let's include error testing for parseDateTime function

manlymonk committed:
Initialize core project files

The project requires essential classes to manage tasks, user interaction, data storage, and command parsing. Currently, these functionalities are missing.

To address this, let's introduce the following classes:

Ui.java: Handles user interactions and message display.
TaskList.java: Manages the list of tasks and task-related operations.
Storage.java: Handles file reading and writing to persist tasks.
Parser.java: Processes user input and executes corresponding commands.
Javen.java: Main entry point of the application.
These additions establish the project's foundational structure for task management and command execution.

manlymonk committed:
Update expected.txt for error testing

The expected.txt file contains reference outputs for error handling tests. Some test cases were outdated or incorrect, leading to inconsistencies in test results.

Let's update expected.txt to reflect the latest expected error messages, ensuring accurate and reliable test verification.

manlymonk committed:
Merge branch 'A-MoreOOP'

manlymonk committed:
Refactor: Organize classes into relevant packages and update runtest.bat

Moved all classes into appropriate packages under the 'javen' namespace, including parser, storage, task, tasklist, and ui. This restructuring improves project organization and maintainability.

Updated the runtest.bat script to reflect the new package structure, ensuring it correctly compiles and tests the application.

manlymonk committed:
refactor: Organize classes into relevant packages

Moved all classes into appropriate packages under the 'javen' namespace, including parser, storage, task, tasklist, and ui. This restructuring improves project organization and maintainability.

manlymonk committed:
refactor: Organize classes into relevant packages and update runtest.bat

Moved all classes into appropriate packages under the 'javen' namespace, including parser, storage, task, tasklist, and ui. This restructuring improves project organization and maintainability.

manlymonk committed:
Merge branch 'A-Packages'

manlymonk committed:
Merge branch 'add-gradle-support'

manlymonk committed:
Test: Add JUnit tests for Parser and TaskList classes

- Added unit tests for Parser class to validate string-to-integer conversion logic
- Added unit tests for TaskList class to check task list functionalities

manlymonk committed:
Merge branch 'A-JUnit'

manlymonk committed:
Change build.gradle: Add Shadow plugin for executable JAR packaging

manlymonk committed:
Merge branch 'A-Jar'

thegrimbee commented on src/main/java/javen/parser/Parser.java:
```diff
@@ -0,0 +1,238 @@
+            if (deadlineParts.length < 2) {
+                System.out.println("""
+                        ________________________________________
+                        Hmm... There's something wrong with your input!
```
I like how descriptive the error messages are!

thegrimbee commented on src/main/java/javen/tasklist/TaskList.java:
```diff
@@ -0,0 +1,240 @@
+     * @param task Task that users add
+     * @return a message to user
+     */
+    public String addTask(Task task) {
```
Great job at being consistent in naming methods with verbs!

thegrimbee commented on src/main/java/javen/tasklist/TaskList.java:
```diff
@@ -0,0 +1,240 @@
+     * @param keyword Keyword input by user.
+     * @return a message to user
+     */
+    public String searchTask(String keyword) {
```
Just a small nitpick, it might be better to name the method findTask instead to be consistent with the find command

thegrimbee commented on src/test/java/javen/parser/ParserTest.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @Test
+    public void checkString_toIntegerTest() {
```
Not sure whether having _  in method name follows the naming guidelines. Might be better to just name it checkStringToIntegerTest()

thegrimbee reviewed (commented):
LGTM, only small nitpicks. Code is very clean and largely follows the guidelines!

manlymonk committed:
Add Checkstyle configuration and fix style issues

The project lacked a Checkstyle configuration, making it difficult
to enforce consistent code style automatically.

Adds a Checkstyle XML configuration file to enforce style rules
Reorders import statements to align with Checkstyle expectations
Fixes minor formatting inconsistencies in affected files

manlymonk committed:
Integrate JavaFX for GUI and refactor classes accordingly

Added JavaFX dependencies for GUI implementation
Created Main.java to initialize and launcher.java to launch the JavaFX application
Adjusted class structures to support GUI-based interactions
Updated image loading paths and fixed resource access issues

This commit sets up the foundation for a JavaFX-based user interface.

manlymonk committed:
Integrate JavaFX for GUI and refactor classes accordingly

Added JavaFX dependencies for GUI implementation
Created Main.java to initialize and launch the JavaFX application
Adjusted class structures to support GUI-based interactions
Updated image loading paths and fixed resource access issues

This commit sets up the foundation for a JavaFX-based user interface.

manlymonk committed:
Merge tag 'Level-10'

manlymonk committed:
Add assertions to enforce preconditions

Insert assertions in key classes to validate inputs and internal state.
This commit adds checks to ensure that parameters (e.g., file paths,
input strings, and task lists) are not null and meet expected criteria.
By asserting preconditions in Storage, Parser, TaskList, and Task, we
catch programming errors earlier during development and document
assumptions about the code's invariants.

manlymonk committed:
Chore: improve code quality and error handling in TaskList

Fixed incorrect Javadoc descriptions for multiple methods
Replaced magic number with constants
Apply SLAP
Made happy path prominent
Improved string formatting for better readability

These changes enhance code maintainability, readability, and correctness.

javentankangle committed:
Merge pull request #1 from javentankangle/A-Assertions

Add assertions to enforce preconditions

javentankangle committed:
Merge pull request #2 from javentankangle/A-CodeQuality

Chore: improve code quality and error handling in TaskList

patil-np commented on src/main/java/javen/tasklist/TaskList.java:
```diff
@@ -0,0 +1,243 @@
+ * A tasklist that handles various tasks
+ */
+public class TaskList {
+    private static final int INVALID_INDEX = -1;
```
I like that you adhered to the coding standards for declaring constant fields!

patil-np commented on src/main/java/javen/parser/Parser.java:
```diff
@@ -0,0 +1,241 @@
+     * @param ui javen.ui to print messages
+     * @param storage javen.storage to save tasks
+     */
+    public String readInput(String input, TaskList taskList, Ui ui, Storage storage) {
```
I like how well-organised this method of your class is! Keep it up.

patil-np commented on src/main/java/javen/parser/Parser.java:
```diff
@@ -0,0 +1,241 @@
+        StringBuilder errorBuilder = new StringBuilder();
+        String details = (parts.length > 1) ? parts[1] : "";
+
+        switch (command) {
```
I like that you remembered the proper coding standards for switch statements!

patil-np reviewed (commented):
I think that your code follows the coding standards well. Good job and keep up the good work!

manlymonk committed:
Create branch for IP progress dashboard

Fix final field in TaskList.java

Tasks in constructor are not final

Task should be final once they are created

Let's make the task final

Task cannot be change after it is final

manlymonk committed:
Fix path for local save in storage.java

File path for storage.java is Duke.txt

There is no relevance to Duke

Let's change the name to Javen.txt

Javen is the project name and should be reflected accordingly

manlymonk committed:
Fix path for local save in javen.java

File path for storage.java is Javen.txt and not Duke.txt

There is no relevance to Duke

Let's change the name to Javen.txt

Javen is the project name and should be reflected accordingly

manlymonk committed:
Add title bar to GUI

Title bar has no name

Title bar should reflect project name so users will know the project name

Let's add the project name to title bar

User will know the application name

manlymonk committed:
Fix bug in TaskList.java

No descript were shown when task was found

User would not understand what is going on

Let's add description after finding for task

Users can understand what is going on

manlymonk committed:
Add user guide

User guide is missing

User guide is important for user to understand how the Javen works

Let's add a user guide

Users can understand how the application work

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Add javadocs for method

Missing javadocs method in Javen.java

Javadocs helps user understand the method

Let's include javadocs for the method

This javadocs helps user understand the method better

manlymonk committed:
Add javadocs for method

Missing javadocs method in Javen.java

Javadocs helps user understand the method

Let's include javadocs for the method

This javadocs helps user understand the method better

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Fix bug in MainWindow.java

Application should close if input contains bye

Application close only if input contains bye

Let's fix the bug and make it such that the application closes when user inputs startswith bye

Application works correctly

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Fix user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide

manlymonk committed:
Add user guide

There are irrelevant comments in user guide

Users can misunderstand the user guide

Let's remove the irrelevant comments

Users will better understand the guide
